ERROR - 2022-01-10 14:04:43 - hdx.utilities.retriever - current exchange rates download failed, using static data fallback_data/currentrates.json!
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hdx/utilities/downloader.py", line 281, in normal_setup
    self.response.raise_for_status()
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 521 Server Error:  for url: https://api.exchangerate.host/latest?base=usd

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hdx/utilities/retriever.py", line 219, in retrieve_json
    self.downloader.download(url, **kwargs)
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hdx/utilities/downloader.py", line 415, in download
    return self.setup(
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hdx/utilities/downloader.py", line 285, in normal_setup
    raise DownloadError(
hdx.utilities.downloader.DownloadError: Setup of Streaming Download of https://api.exchangerate.host/latest?base=usd failed!
ERROR - 2022-01-10 14:33:41 - iati.activity - Activity FI-3-2020-2020201403 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:41 - iati.activity - Activity FI-3-2020-2020201234 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:41 - iati.activity - Activity FI-3-2020-2020201168 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:41 - iati.activity - Activity FI-3-2020-2020201166 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:42 - iati.activity - Activity FI-3-2020-2020201165 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:42 - iati.activity - Activity FI-3-2020-2020201160 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:42 - iati.activity - Activity FI-3-2020-2020200801 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:42 - iati.activity - Activity FI-3-2020-2019191455 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:42 - iati.activity - Activity FI-3-2020-2019191454 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:42 - iati.activity - Activity FI-3-2020-2019191453 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:42 - iati.activity - Activity FI-3-2020-2019191451 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:42 - iati.activity - Activity FI-3-2020-2019191448 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
ERROR - 2022-01-10 14:33:42 - iati.activity - Activity FI-3-2020-2019191446 is_strict call failed!
Traceback (most recent call last):
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/activity.py", line 69, in is_strict
    or Lookups.checks.has_desired_tag(self.dactivity.tags)
  File "/home/runner/work/hdx-scraper-iati-viz/hdx-scraper-iati-viz/iati/covid_checks.py", line 24, in has_desired_tag
    if tag.vocabulary == "99" and tag.code.upper() == "COVID-19":
AttributeError: 'NoneType' object has no attribute 'upper'
